src/driver/types.rs

```diff
@@ -15,24 +18,54 @@ pub struct HeadInfo {
     pub sequence_number: u64,
 }
 
-impl TryFrom<Block<Transaction>> for HeadInfo {
-    type Error = eyre::Report;
+impl HeadInfo {
+    /// Returns the head info from the given L2 block and the system config.
+    /// The config is used to check whether the block is subject to the Ecotone hardfork
+    /// (which changes the way the head info is constructed from the block).
+    pub fn try_from_l2_block(config: &Config, l2_block: Block<Transaction>) -> Result<Self> {
+        if config
+            .chain
+            .is_ecotone_but_not_first_block(l2_block.timestamp.as_u64())
```

Comment:
Is the idea here that the attributes deposited contract is not upgraded until the second ecotone block, so we should use the old version to fetch the data for the first block?

Reply:
Exactly, as `setL1BlockValuesEcotone` is not guaranteed to exist on the fork activation block. 
specs: https://specs.optimism.io/protocol/deposits.html#l1-attributes---ecotone